### PR TITLE
COMPAT putBucketACL return InvalidArgument for bad ID

### LINKED ISSUE #365: COMPAT: putBucketACL should return InvalidArgument if invalid canonical user ID
Should return `InvalidArgument` if attempting to put a bucket and grant ACL access with a canonical user ID that is not 64 characters long hex string.

On **ceph/s3-tests**, this test fails:
* `FAIL: s3tests.functional.test_s3.test_bucket_acl_grant_nonexist_user`

### PR DESCRIPTION
**Pull request by electrachong**
FIXES #365 

Had to modify some existing unit tests to comply with new requirements.

### CODE REVIEW

**Comment by electrachong** [comment]
@ironman-machine check it please

**Comment by electrachong** [comment]
Actually the string should be a hex string, not any alphanumeric string, so I fixed that and added a commit.

File: lib/utilities/aclUtils.js [hunk 1]
```diff
@@ -171,6 +171,10 @@ aclUtils.parseGrant = function parseGrant(grantHeader, grantType) {
     });
 };
 
+aclUtils.isValidCanonicalId = function isValidCanonicalId(canonicalID) {
+    return /^[a-f0-9]{64}$/.test(canonicalID);
```
Comment (alexandre-merle) [inline lib/utilities/aclUtils.js]: it is always lowercase ?

Comment (electrachong) [inline lib/utilities/aclUtils.js]: I was wondering about this myself, AWS only generates canonical IDs with a-f in lowercase but I just tested using aws s3api client and it also accepts uppercase letters as a valid id. Thanks, I updated the code.

File: tests/unit/helpers.js [hunk 2]
```diff
@@ -43,6 +43,8 @@ export function makeAuthInfo(accessKey) {
             + 'cd47ef2be',
         accessKey2: '79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7'
             + 'cd47ef2bf',
+        accessKey3: '79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7'
```
Comment (LaurenSpiegel) [inline tests/unit/helpers.js]: where is this used?

Comment (electrachong) [inline tests/unit/helpers.js]: This was added to make sure tests in [tests/unit/api/objectDelete](https://github.com/scality/S3/blob/master/tests/unit/api/objectDelete.js) pass with the new checks on the IDs. The objectDelete tests use the makeAuthInfo helper function to generate canonical IDs, including makeAuthInfo('accessKey3'), but accessKey3 did not previously map to a valid canonical ID.

**Comment by electrachong** [comment]
@LaurenSpiegel I added a functional test and modified the authInfo helper function.

File: tests/functional/aws-node-sdk/test/bucket/putACL.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import assert from 'assert';
+
+import withV4 from '../support/withV4';
+import BucketUtility from '../../lib/utility/bucket-util';
+
+const bucketName = 'putbucketaclfttest';
+
+describe.only('PUT Bucket ACL', () => {
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/bucket/putACL.js]: scratch the .only

Comment (electrachong) [inline tests/functional/aws-node-sdk/test/bucket/putACL.js]: removed it, thanks

**Comment by electrachong** [comment]
@ironman-machine what do you say?

**Comment by electrachong** [comment]
@ironman-machine now what do you say? :P

**Comment by electrachong** [comment]
@ironman-machine try with this branch please
SCALITY_VAULT_BRANCH=COMPAT/AclInvalidUserID

**Comment by alexandre-merle** [comment]
@ironman-machine go alone, checking for previous format compat :)

**Comment by electrachong** [comment]
end-to-end passed, squashed commits and rebased

**Pull request merged**